src/fides/api/models/datasetconfig.py

```diff
@@ -403,3 +410,28 @@ def validate_dataset_reference(
         raise ValidationError(
             f"Unknown field '{dataset_reference.field}' in dataset '{dataset_config.fides_key}' referenced by external reference"
         )
+
+
+def validate_masking_strategy_override(dataset: Dataset) -> None:
```

Comment:
We should add a docstring explaining the need for this validation. When we receive a privacy request, we look at the policies to determine which masking strategies we need to generate and cache secret keys. For now, we don't want to add the extra scope of also having to scan the datasets for masking overrides. You can site the `cache_data` function in https://github.com/ethyca/fides/blob/field_level_masking/src/fides/api/service/privacy_request/request_service.py#L59